api/app/geotiff_from_stac_api.py

```diff
@@ -66,10 +65,11 @@ def generate_geotiff_from_stac_api(
     # Add the actual outputed band info to the filename
     final_band = list(collections_dataset.keys())[0]
     band_suffix = "_" + final_band if (final_band != "band") else ""
-    file_path = f"{collection}{band_suffix}_{date or 'no_date'}.tif"
+    bbox_hash = hashlib.sha256(str(bbox).encode()).hexdigest()[:8]
+    file_path = f"{collection}{band_suffix}_{bbox_hash}_{date or 'no_date'}.tif"
 
     try:
-        write_cog(collections_dataset[final_band], file_path)
+        write_cog(collections_dataset[final_band], file_path, overwrite=True)
     except Exception as e:
```

Comment:
Is there a way to filter exceptions just a bit more specifically?

Reply:
Not really sure how but open to suggestions :)